price-feeder/oracle/oracle.go

```diff
@@ -185,16 +188,33 @@ func (o *Oracle) SetPrices(ctx context.Context, acceptList oracletypes.DenomList
 				return err
 			}
 
+			candles, err := priceProvider.GetCandlePrices(acceptedPairs...)
+			if err != nil {
+				telemetry.IncrCounter(1, "failure", "provider")
+				return err
+			}
+
 			// flatten and collect prices based on the base currency per provider
 			//
 			// e.g.: {ProviderKraken: {"ATOM": <price, volume>, ...}}
 			mtx.Lock()
-			for _, cp := range acceptedPairs {
+			for _, pair := range acceptedPairs {
```

Comment:
What if we miss a pair for a latest trade but not for a candle, we'll prematurely return and never call `providerCandles[providerName][pair.Base] = cp`. Is this expected? Should we unlock and error only iff we dont get a candle and trade?

Reply:
Switched it to :
```golang
tp, pricesOk := prices[pair.String()]
cp, candlesOk := candles[pair.String()]
if pricesOk {
	providerPrices[providerName][pair.Base] = tp
}
if candlesOk {
	providerCandles[providerName][pair.Base] = cp
}

if !pricesOk && !candlesOk {
	mtx.Unlock()
	return fmt.Errorf("failed to find any exchange rates in provider response")
}
```